src/dut_power.rs

```diff
@@ -173,6 +173,49 @@ pub struct DutPwrThread {
     tick: Arc<AtomicU32>,
 }
 
+struct MedianFilter<const N: usize> {
+    history: [f32; N],
+    index: usize,
+    ready: bool,
+}
+
+impl<const N: usize> MedianFilter<N> {
+    pub fn new() -> Self {
+        Self {
+            history: [f32::NAN; N],
+            index: 0,
+            ready: false,
+        }
+    }
+
+    /// Return the median of the N last values added or None if less than N
+    /// values were stepped in yet.
+    ///
+    /// Returns the mean of the two center most entries if N is even.
+    pub fn step(&mut self, val: f32) -> Option<f32> {
+        self.history[self.index] = val;
+        self.index = (self.index + 1) % N;
+        self.ready |= self.index == 0;
+
+        if self.ready {
+            let sorted = {
+                let mut sorted = [0.0; N];
+                sorted.clone_from_slice(&self.history);
+                sorted.sort_unstable_by(f32::total_cmp);
+                sorted
+            };
+
+            if N % 2 == 0 {
+                Some((sorted[N / 2 - 1] + sorted[N / 2]) / 2.0)
+            } else {
+                Some(sorted[N / 2])
```

Comment:
Perhaps enforce an odd length to avoid the complexity?

Reply:
When eyeballing how we wanted this filter to work we always had a filter with four values in mind.
*"Take four values, throw away the largest and smallest one and average the other two"*,
so that's what I've implemented.

We could go to another filter length if we want. Four was just our first intuition.